src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,230 @@
 package core.basesyntax;
 
+import java.util.Collection;
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
     @Override
     public void add(T value) {
+        linkedLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            linkedLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
+    }
+
+    void linkBefore(T e, Node<T> succ) {
```

Comment:
```suggestion
    private void linkBefore(T e, Node<T> succ) {
```

Private methods and classes should be after public ones in your class. Move the linkBefore method after all public methods.